test/tests/bootstrap/test_api20_os_bootstrap_parallel_local.py

```diff
@@ -0,0 +1,292 @@
+                "smbUser": "onrack",
+                "version": "2012",
+                "workflow": "Graph.InstallWindowsServer"
+            }
+        ],
+
+The OS repos are to be installed under 'on-http/static/http' directory reflecting the paths above.
+These can be files, links, or nfs mounts to remote repos in the following dirs:
+
+on-http/static/http/windowsServer2012 -- requires Samba share on RackHD server
+on-http/static/http/repo/centos/6.5
+on-http/static/http/repo/centos/7.0
+on-http/static/http/repo/rhel/7.0
+on-http/static/http/repo/suse/42.1
+on-http/static/http/repo/esxi/5.5
+on-http/static/http/repo/esxi/6.0
+on-http/static/http/repo/winpe
+on-http/static/http/repo/coreos/899.17.0
+
+'''
+
+import fit_path  # NOQA: unused import
+from nose.plugins.attrib import attr
+import fit_common
+import flogging
+log = flogging.get_loggers()
+
+# This gets the list of nodes
+NODECATALOG = fit_common.node_select()
+
+# dict containing bootstrap workflow IDs and states
+NODE_STATUS = {}
+
+# collect repo information from config files
+OSLIST = fit_common.fitcfg()["install-config"]["os-install"]
+
+# download RackHD config from host
+rackhdconfig = fit_common.rackhdapi('/api/2.0/config')['json']
+rackhdhost = "http://" + str(rackhdconfig['apiServerAddress']) + ":" + str(rackhdconfig['apiServerPort'])
+
+
+# this routine polls a task ID for completion
+def wait_for_task_complete(taskid, retries=60):
+    for dummy in range(0, retries):
+        result = fit_common.rackhdapi("/api/2.0/workflows/" + taskid)
+        if result['status'] != 200:
+            log.error(" " + result['text'])
+            return False
+        if result['json']['status'] == 'running' or result['json']['status'] == 'pending':
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+            fit_common.time.sleep(30)
+        elif result['json']['status'] == 'succeeded':
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+            log.info_5(" " + result['text'])
+            return True
+        else:
+            log.error(" " + result['text'])
+            return False
+    log.error(" Workflow Timeout: " + result['text'])
+    return False
+
+
+# helper routine to return the task ID associated with the running bootstrap workflow
+def node_taskid(workflow, version, kvm):
+    for entry in NODE_STATUS:
+        if NODE_STATUS[entry]['workflow'] == workflow \
+                and str(version) in NODE_STATUS[entry]['version'] \
+                and NODE_STATUS[entry]['kvm'] == kvm:
+            return NODE_STATUS[entry]['id']
+    return ""
+
+
+# Match up tests to node IDs to feed skip decorators
+index = 0  # node index
+for item in OSLIST:
+    if index < len(NODECATALOG):
+        NODE_STATUS[NODECATALOG[index]] = \
+            {"workflow": item['workflow'], "version": item['version'], "kvm": item['kvm'], "id": "Pending"}
```

Comment:
There is no workflow submitted yet, but this key needs to be initialized so I used 'pending'. Any other ideas?